src/reducers/loginReducers.js

```diff
@@ -0,0 +1,23 @@
+import initialState from '../store/initialState';
+import { loginType } from '../actions/loginActions';
+
+/**
+ * 
+ * @param {*} state - The initial user state
+ * @param {*} action - The action payload
+ * @returns {Object} - The current user state 
+ */
+const LoginReducer = (state = initialState.user, action) => {
+  switch (action.type) {
+    case loginType.success:
+      return { ...state, ...action.payload };
```

Comment:
#### Nitpick
- So if the state is the user initial state ie. { id: 1, token: iu89ert } and action payload { id: 1, token: 99ragdf }, this implementation with the spread operator will resilt in as 
```
{
 id: 1, 
 token: iu89ert,
 id: 1, 
 token: 99ragdf
}

```
I want to understand if this the desired result

Reply:
Hello Audax.

Thank you for the feedback. No, this isn't the desired effect. if initial state is {id: 1, token: iu89ert} and payload has {id: 1, token: 99radgf}. the resulting state will be:
```
{
  id: 1,
  token: 99radgf
}
```